### ISSUE TITLE
[bitnami/wordpress-nginx] 6.6.1 is not compatible with bitnami/mariadb:11.0.4

### ISSUE THREAD

**Issue by timofey-drozhzhin** [bitnami/containers#70392]
### Name and Version

bitnami/wordpress-nginx:6.6.1

### What architecture are you using?

arm64

### What steps will reproduce the bug?

Create `compose.yaml` with the following:
```yaml
# Copyright Broadcom, Inc. All Rights Reserved.
# SPDX-License-Identifier: APACHE-2.0

version: '2'
services:
  mariadb:
    image: docker.io/bitnami/mariadb:11.0.4
    volumes:
      - 'mariadb_data:/bitnami/mariadb'
    environment:
      # ALLOW_EMPTY_PASSWORD is recommended only for development.
      - MARIADB_ROOT_USER=root
      - MARIADB_DATABASE=bitnami_wordpress
      - MARIADB_ROOT_PASSWORD=password
  wordpress:
    image: docker.io/bitnami/wordpress-nginx:6.6.1
    ports:
      - '80:8080'
      - '443:8443'
    volumes:
      - 'wordpress_data:/bitnami/wordpress'
    depends_on:
      - mariadb
    environment:
      # ALLOW_EMPTY_PASSWORD is recommended only for development.
      - WORDPRESS_AUTO_UPDATE_LEVEL=none
      - WORDPRESS_DATABASE_HOST=mariadb
      - WORDPRESS_DATABASE_PORT_NUMBER=3306
      - WORDPRESS_DATABASE_USER=root
      - WORDPRESS_DATABASE_PASSWORD=password
      - WORDPRESS_DATABASE_NAME=bitnami_wordpress
      - BITNAMI_DEBUG=true
      - MYSQL_CLIENT_DATABASE_HOST=mariadb
      - MYSQL_CLIENT_DATABASE_PORT_NUMBER=3306
      - MYSQL_CLIENT_DATABASE_ROOT_PASSWORD=password
      - MYSQL_CLIENT_DATABASE_ROOT_USER=root
      - MYSQL_CLIENT_CREATE_DATABASE_CHARACTER_SET=utf8mb4
      - MYSQL_CLIENT_CREATE_DATABASE_COLLATE=utf8mb4_unicode_ci
      - MYSQL_CLIENT_CREATE_DATABASE_NAME=bitnami_wordpress
      - MYSQL_CLIENT_ENABLE_SSL=false
      - MYSQL_CLIENT_ENABLE_SSL_WRAPPER=false
      - MYSQL_CLIENT_FLAVOR=mysql
      - WORDPRESS_VERIFY_DATABASE_SSL=false
volumes:
  mariadb_data:
    driver: local
  wordpress_data:
    driver: local
```

Then start the instance
```
docker compose down -v
docker compose up
```

### What is the expected behavior?

Expecting WordPress to Start

### What do you see instead?

```
wordpress-1  | ERROR 2026 (HY000): TLS/SSL error: SSL is required, but the server does not support it
wordpress-1  | wordpress 03:05:11.16 DEBUG ==> Executing SQL command:
wordpress-1  | SELECT 1
wordpress-1  | /opt/bitnami/mysql/bin/mysql: Deprecated program name. It will be removed in a future release, use '/opt/bitnami/mysql/bin/mariadb' instead
```

<img width="1399" alt="Screenshot 2024-07-27 at 10 05 30 PM" src="https://github.com/user-attachments/assets/4434bb2d-27aa-490f-aa9e-8b5145f3d346">

### Additional information

Running the same compose.yaml file with `bitnami/wordpress-nginx:6.6.0` works fine.

This issue may be related to #70007

**Comment by timofey-drozhzhin** [comment]
So to clarify:
`bitnami/wordpress-nginx:6.6.1` is not compatible with `bitnami/mariadb:11.0.4`
`bitnami/wordpress-nginx:6.6.0` is compatible with `bitnami/mariadb:11.0.4`
`bitnami/wordpress-nginx:6.6.1` is compatible with `bitnami/mariadb:11.4`

**Comment by gongomgra** [comment]
Hi @timofey-drozhzhin,

As mentioned in the other thread, and according to our [GitHub history](https://github.com/bitnami/containers/commit/a8f6e3e4270929202a4197372427c3660389213a), the MariaDB tag was updated to `11.4` two weeks ago, which seems to fix the issue that you are sharing here. Apart from that I have removed all the images on my computer and have started the WordPress NGINX compose file a couple of times without any issues.

Can you check that you are using the latest docker-compose file?

**Comment by bashbang** [comment]
I too was having issues.  Here are the rest results of what I tried:

bitnami/wordpress-nginx:6.6.1 does not connect to bitnami/mariadb:11.2.4 at init.

ERROR from Wordpress Image:
mariadb-1  | 2024-07-29 23:23:01 3 [Warning] Aborted connection 3 to db: 'unconnected' user: 'unauthenticated' host: '172.21.0.3' (This connection closed normally without authentication)

However:
bitnami/wordpress-nginx:6.6.0 does connect to bitnami/mariadb:11.2.4 without issue.
bitnami/wordpress-nginx:6.6.1 does connect to bitnami/mariadb:11.4.2 without issue.

**Comment by timofey-drozhzhin** [comment]
@bashbang I can confirm, bitnami/wordpress-nginx:6.6.1 cannot connect to bitnami/mariadb:11.2.4 also.
```
wordpress-1  | ERROR 2026 (HY000): TLS/SSL error: SSL is required, but the server does not support it
mariadb-1    | 2024-07-29 23:44:42 4 [Warning] Aborted connection 4 to db: 'unconnected' user: 'unauthenticated' host: '172.25.0.3' (This connection closed normally without authentication)
```

@gongomgra - you're correct in that `bitnami/wordpress-nginx:6.6.1` works with `bitnami/mariadb:11.4`, but it's no longer compatible with anything below v11.4.

The reason I mention this, is because someone like us, that was using mariadb v11.0.4, didn't expect a minor release of WordPress (`6.6.0` => `6.6.1`) to break our website.

**Comment by renilthomas** [comment]
A minor version release should not have introduced such an incompatibility. And, this clearly contradicts the actual wordpress compatibility [matrix](https://make.wordpress.org/hosting/handbook/compatibility/#wordpress-php-mysql-mariadb-versions)

**Comment by gongomgra** [comment]
I'm afraid I have not being able to reproduce your issue. I have been able to start WordPress from scratch (no persisted data) with MariaDB `10.4` and `11.0.4` tags.

```
$ git diff .
diff --git a/bitnami/wordpress/6/debian-12/docker-compose.yml b/bitnami/wordpress/6/debian-12/docker-compose.yml
index c47931cc59f..b58f1115193 100644
--- a/bitnami/wordpress/6/debian-12/docker-compose.yml
+++ b/bitnami/wordpress/6/debian-12/docker-compose.yml
@@ -4,7 +4,7 @@
 version: '2'
 services:
   mariadb:
-    image: docker.io/bitnami/mariadb:11.4
+    image: docker.io/bitnami/mariadb:11.0.4
     volumes:
       - 'mariadb_data:/bitnami/mariadb'
     environment:

$ docker-compose up -d && docker-compose logs -f wordpress
[+] Running 2/2
 ✔ mariadb Pulled                                                                                                                    25.2s
   ✔ 2446c9cacd8b Pull complete                                                                                                      21.8s
[+] Running 5/5
 ✔ Network debian-12_default          Created                                                                                         0.2s
 ✔ Volume "debian-12_wordpress_data"  Created                                                                                         0.0s
 ✔ Volume "debian-12_mariadb_data"    Created                                                                                         0.0s
 ✔ Container debian-12-mariadb-1      Started                                                                                         2.0s
 ✔ Container debian-12-wordpress-1    Started                                                                                         1.6s

wordpress-1  | wordpress 09:34:26.11 INFO  ==>
wordpress-1  | wordpress 09:34:26.12 INFO  ==> Welcome to the Bitnami wordpress container
(...)
wordpress-1  | wordpress 09:34:26.17 INFO  ==> ** Starting WordPress setup **
(...)
wordpress-1  | wordpress 09:34:32.28 INFO  ==> Trying to connect to the database server
wordpress-1  | wordpress 09:34:37.35 INFO  ==> Configuring WordPress with settings provided via environment variables
wordpress-1  | wordpress 09:34:39.67 INFO  ==> Installing WordPress
wordpress-1  | wordpress 09:34:45.91 INFO  ==> Persisting WordPress installation
wordpress-1  | wordpress 09:34:56.06 INFO  ==> ** WordPress setup finished! **
wordpress-1  |
wordpress-1  | wordpress 09:34:56.10 INFO  ==> ** Starting Apache **
```

Can it be related to any other change you have on your side?

**Comment by renilthomas** [comment]
With `bitnami/wordpress-nginx:6.6.1` I can reproduce the issue against  `docker.io/bitnami/mariadb:10.6.16` or `11.0.4`, while `6.6.0` works fine. Are you not able to reproduce this with the nginx image mentioned here, I see from the logs that you may be using a different wordpress image?

**Comment by floek** [comment]
Same issue here: mariadb:11.3.2 and bitnami/wordpress-nginx:6.6.1 ist broken. Upgraded to mariadb:11.4.2 and now its back up and running.

**Comment by timofey-drozhzhin** [comment]
@gongomgra you will need to use the `compose.yaml` values that I provided in my first comment. You will not be able to reproduce the bug using the original `bitnami/wordpress/6/debian-12/docker-compose.yml` file, because the bug affects instances where ALLOW_EMPTY_PASSWORD is not set to true.

See my screencast https://www.berrycast.com/conversations/df02b5b4-4fb1-509c-87ce-d86dff604d89

**Comment by gongomgra** [comment]
@renilthomas yes, you are right, the log I shared was for WordPress image instead of WordPess-NGINX, but the image is basically the same just with a different the webserver. They are both affected by the same issue.

@timofey-drozhzhin thanks for the information on how to reproduce it. I think I have found the issue and I'm working on a fix for it. 

I have found that we increased MySQL/MariaDB client minor version from `11.3` to `11.4` starting with image `wordpress-nginx:6.6.1` (see [pr/69975](https://github.com/bitnami/containers/pull/69975)). According to [MariaDB 11.4 changelog](https://mariadb.com/kb/en/mariadb-11-4-1-changelog/) they have enabled SSL by default for the client (unless the `--skip-ssl` flag is provided). I have also checked we are not having this into account on our side when the `MYSQL_CLIENT_ENABLE_SSL=no` environment variable is provided, so I will work on fixing this.

As a test, I have created a custom WordPress-NGINX `6.6.1` image including the patch below and I have been able to start it against MariaDB tag `11.0.4` without issues when a database password is configured.

```
diff --git a/bitnami/wordpress-nginx/6/debian-12/rootfs/opt/bitnami/scripts/libmysqlclient.sh b/bitnami/wordpress-nginx/6/debian-12/rootfs/opt/bitnami/scripts/libmysqlclient.sh
index ad32c9c374b..1c76089d63f 100644
--- a/bitnami/wordpress-nginx/6/debian-12/rootfs/opt/bitnami/scripts/libmysqlclient.sh
+++ b/bitnami/wordpress-nginx/6/debian-12/rootfs/opt/bitnami/scripts/libmysqlclient.sh
@@ -934,6 +934,8 @@ mysql_client_extra_opts() {
             value="$(mysql_client_env_value "SSL_${key^^}_FILE")"
             [[ -n "${value}" ]] && opts+=("--ssl-${key}=${value}")
         done
+    else
+        opts+=("--skip-ssl")
     fi
     echo "${opts[@]:-}"
 }
 ```

Hope it works as a workaround for you as well while we work on a new release including the fix.

**Comment by gongomgra** [comment]
Hi all,

We have just released a new revision of our WordPress image including the fix for this issue. 

```
docker pull bitnami/wordpress:6.6.1-debian-12-r3
```

You can check the changes included in this new image at https://github.com/bitnami/containers/commit/57ad942fe94980bc7881df57561b0e3cda460a66

**Comment by chuckreynolds** [comment]
Will the patch update in `bitnami/wordpress-nginx:6.6.1` too?

**Comment by carrodher** [comment]
Yes, that's a rolling tag pointing to the latest immutable tag

**Comment by renilthomas** [comment]
@gongomgra in the `6.6.1` image I see there is a different file permissions for `/opt/bitnami/scripts/wordpress` as a result the container is exiting because of 
`sed: couldn't open temporary file /opt/bitnami/scripts/wordpress/sed3sDh2n: Permission denied `

in 6.6.0 
```
$ ls -ld /opt/bitnami/scripts/wordpress
drwxrwxr-x 3 root root 4096 Jul 18 16:20 /opt/bitnami/scripts/wordpress
```

while in 6.6.1
```
$ ls -ld /opt/bitnami/scripts/wordpress
drwxr-xr-x 3 root root 4096 Aug 12 23:06 /opt/bitnami/scripts/wordpress
```

**Comment by juan131** [comment]
Hi @renilthomas 

Thanks for raising this issue, but should try to stick to the incompatibilities with MariaDB due to TLS changes topic in order to avoid confusion. 

Could you please open a different issue describing the change in the permissions? Thanks in advance.

**Comment by renilthomas** [comment]
sure @juan131 - I have raised [70940](https://github.com/bitnami/containers/issues/70940)

**Comment by juan131** [comment]
Fixed on `bitnami/wordpress-nginx:6.6.1-debian-12-r3` as explained at https://github.com/bitnami/containers/issues/70392#issuecomment-2265486447
